DockWin.ahk

```diff
@@ -0,0 +1,207 @@
+;DockWin v0.5 - Save and Restore window positions when docking/undocking (using hotkeys) over all Windows 10 Desktops
+; Paul Troiano, 6/2014
+; Updated by Ashley Dawson 7/2015
+; Updated by Robert Niebsch 10/2016
+;
+; Hotkeys: Capslock + 0 to save; Capslock + 9 to load
+
+class DockWinClass
+{
+
+    SaveAllWindowsFunctionName := "SaveAllWindows"
+    LoadAllWindowsFunctionName := "LoadAllWindows"
+    
+	
+	__new()
+    {  
+        this.WindowMover := new JPGIncWindowMoverClass()
+		this.desktopMapper := new DesktopMapperClass(new VirtualDesktopManagerClass())
+		this.monitorMapper := new MonitorMapperClass()  
+        return this
+    }
+    
+; goes through file
+; title of line in file is matched with open windows
+LoadAllWindows()

```

Comment:
This function has a couple of parts that are re-used in SaveAllWindows. Perhaps you should pull them out into their own functions. 

I think that an ini file might be a better format for the settings file. That way you don't need to parse it at all. see https://autohotkey.com/docs/commands/IniRead.htm and it's related sections.

For example you could do something like
sectionHeader := SectionHeader() 
IniRead, section, % Filename, % sectionHeader
if(! section) {
 msgbox, no section found
}

you could have each saved window as a numeric key and then loop over all the keys using something like this
while(true)
{
key := A_index
 IniRead, value, % Filename, % sectionHeader, % key
if(value) {
;do stuff
} else {
break
}
}